blockly-core/core/ui/block_space/auto_scroll.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Visual Blocks Editor
+ *
+ * Copyright 2011 Google Inc.
+ * http://blockly.googlecode.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Represents an active, update-able automatic scroll behavior.
+ * @author fraser@google.com (Neil Fraser)
```

Comment:
Doesn't look like you pulled this from upstream, might want to update the authorship metadata.